docs/developer-guide/meta/dist-lock.md

```diff
@@ -0,0 +1,57 @@
+# Distributed Lock
+
+## Introduction
+
+Meta provides the ability to distribute lock externally via gRPC.
+
+## How to use
+
+A meta client implemented by rust is provided in [meta-client][1] crate.
+
+[1]: https://github.com/GreptimeTeam/greptimedb/tree/develop/src/meta-client
+
+For example:
+
+```rust
+async fn do_some_work(meta_client: MetaClient) {
+    let name = b"lock_name".to_vec();
+    let expire_secs = 60;
+
+    let lock_req = LockRequest { name, expire_secs };
+
+    let lock_result = meta_client.lock(lock_req).await.unwrap();
+    let key = lock_result.key;
+
+    info!("do some work, take 3 seconds");
+    tokio::time::sleep(Duration::from_secs(3)).await;
+
+    let unlock_req = UnlockRequest { key };
+
+    meta_client.unlock(unlock_req).await.unwrap();
+    info!("unlock success!");
+}
+```
+
+More examples [here][2].
+
+[2]: https://github.com/GreptimeTeam/greptimedb/blob/develop/src/meta-client/examples/lock.rs
+
+For using other languages, since gRPC is cross-language, you can easily implement language-specific client.
+
+You can find the proto definition [here][3].
+
+[3]: https://github.com/GreptimeTeam/greptime-proto/blob/main/proto/greptime/v1/meta/lock.proto
+
+Pay attention to the following points:
+
+1. If the expiration time of distribute lock is exceeded and currently holds the lock, the lock is automatically released.
```

Comment:
```suggestion
1. The distributed lock will be automatically released if the expiration time which holds it, is exceeded.
```

Reply:
@apdong2022 it's not the expiration time holding the lock but the caller. A caller will request a lock with an expiration time, and the caller have to release the lock in time, otherwise the lock will be released by itself.